src/components/tabController/FadedScrollView.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {useCallback} from 'react';
+import {ViewProps, ScrollView, ScrollViewProps, NativeSyntheticEvent, NativeScrollEvent} from 'react-native';
+import Fader from '../fader';
+import withScrollEnabler, {WithScrollEnablerProps} from '../../commons/withScrollEnabler';
+import withScrollReached, {WithScrollReachedProps} from '../../commons/withScrollReached';
+import forwardRef, {ForwardRefInjectedProps} from '../../commons/forwardRef';
+import {Constants} from '../../helpers';
+
+export type FadedScrollViewProps = ViewProps &
+  ScrollViewProps & {
+    children?: React.ReactNode | React.ReactNode[];
+  };
+
+type ScrollReachedProps = FadedScrollViewProps & WithScrollReachedProps;
+type ScrollEnabledProps = ScrollReachedProps & WithScrollEnablerProps;
+type Props = ScrollEnabledProps & ForwardRefInjectedProps;
+
+const FADER_SIZE = 76;
+
+const FadedScrollView = (props: Props) => {
+  const {scrollEnablerProps, scrollReachedProps, children, onScroll: propsOnScroll, ...other} = props;
+  const showLeft =
+    scrollEnablerProps.scrollEnabled &&
+    (Constants.isRTL ? !scrollReachedProps.isScrollAtEnd : !scrollReachedProps.isScrollAtStart);
+  const leftPosition = Constants.isRTL ? Fader.position.RIGHT : Fader.position.LEFT;
+  const showRight =
+    scrollEnablerProps.scrollEnabled &&
+    (Constants.isRTL ? !scrollReachedProps.isScrollAtStart : !scrollReachedProps.isScrollAtEnd);
```

Comment:
if Fader will support RTL out of the box, does that mean we don't to check RTL here?

Reply:
Turns out: names matter - deprecated LEFT\RIGHT into START\END - made the code much simpler.
Android is still bugged, will be handled in the relevant ticket.